vmd-api/openapi.yaml

```diff
@@ -0,0 +1,324 @@
+openapi: 3.1.0
+info:
+  contact:
+    email: nikita.bazulin@unikie.com
+  description: OpenAPI specification for virtual machine management on Ghaf platform
+  license:
+    name: Apache License 2.0
+    identifier: Apache-2.0
+  title: Virtual Machine Management API of the Ghaf platform - OpenAPI
+  version: 1.0.1
+externalDocs:
+  description: Find out more about the Ghaf platform
+  url: https://github.com/tiiuae/ghaf
+servers:
+- description: The primary server
+  url: https://localhost/api/v1
+- description: The test server
+  url: https://localhost/api/test/v1
+tags:
+- description: Operations of virtual machines
+  name: vm
+paths:
+  /vm/list:
+    get:
+      description: Gets list of IDs for all virtual machines
+      operationId: getVmList
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                items: {}
+          description: List of IDs for all virtual machines
+        default:
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+          description: error payload
+      summary: Get list IDs for all virtual machines
+      tags:
+      - vm
+  /vm/info/{id}:
+    get:
+      description: Gets the virtual machine information by ID
+      operationId: getVmInfoById
+      parameters:
+      - description: The ID of the virtual machine
+        explode: false
+        in: path
+        name: id
+        required: true
+        schema: {}
+        style: simple
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/VirtualMachineInfo'
+          description: successful operation
+        "400":
+          description: Invalid virtual machine ID
+        "404":
+          description: Virtual machine ID is valid but not found
+        default:
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+          description: error payload
+      summary: Gets virtual machine info by ID
+      tags:
+      - vm
+  /vm/{action}/{id}:
```

Comment:
I believe my previous comment on mapping the enum verb would not change the spec here, only the implementation where the `{action}` should be parsed before match.

Reply:
Let's push test-client related things into their own PR/issue